config/rbac/role.yaml

```diff
@@ -34,6 +34,18 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
```

Comment:
Does the operator need all the API access?

Reply:
The core api includes 
```
kubectl api-resources --api-group=''
NAME                     SHORTNAMES   APIVERSION   NAMESPACED   KIND
bindings                              v1           true         Binding
componentstatuses        cs           v1           false        ComponentStatus
configmaps               cm           v1           true         ConfigMap
endpoints                ep           v1           true         Endpoints
events                   ev           v1           true         Event
limitranges              limits       v1           true         LimitRange
namespaces               ns           v1           false        Namespace
nodes                    no           v1           false        Node
persistentvolumeclaims   pvc          v1           true         PersistentVolumeClaim
persistentvolumes        pv           v1           false        PersistentVolume
pods                     po           v1           true         Pod
podtemplates                          v1           true         PodTemplate
replicationcontrollers   rc           v1           true         ReplicationController
resourcequotas           quota        v1           true         ResourceQuota
secrets                               v1           true         Secret
serviceaccounts          sa           v1           true         ServiceAccount
services                 svc          v1           true         Service
```
AFAIK we are using all the core api group (PVC/PV will come down the line when model server is incorporate)
@sthaha please confirm

Reply:
Yes, we will need access to PVC but do  note this PR does not  address the RBAC used. I have kept it the same. IIUC, we shouldn't need `get` since `list` and `watch` is what controller-runtime uses to build its cache. This is a minor detail we can address later.